src/main/java/org/gelecekbilimde/scienceplatform/auth/service/RoleService.java

```diff
@@ -0,0 +1,19 @@
+package org.gelecekbilimde.scienceplatform.auth.service;
+
+import org.gelecekbilimde.scienceplatform.common.model.response.Response;
+import org.gelecekbilimde.scienceplatform.auth.model.request.RoleChangeRequest;
+import org.gelecekbilimde.scienceplatform.auth.model.response.UserRoleResponse;
+
+import java.util.List;
+
+public interface RoleService {
+	UserRoleResponse userRoletoAuthorRoleRequest();
+
+	UserRoleResponse makeUserToAuthor(RoleChangeRequest request);
+
+	List<UserRoleResponse> getAllUserRoletoAuthorRoleRequest();
+
+	Response deleteUserRoletoAuthorRoleRequest(RoleChangeRequest request);
+
+	UserRoleResponse makeUserToAdmin(RoleChangeRequest request);
+}
```

Comment:
```suggestion
	UserRoleResponse makeUserToAdmin(RoleChangeRequest request);
	
}
```

Reply:
Service kısmı için de controllerdaki düşüncemin aynısını söyleyeceğim. Author ve moderatör yapma durumu tek bir yerden yönetilebilir.  Bu durumu updateUserRole tarzı bir method ile halledebiliriz. Sadece Admin erişebileceği için adminin verdiği role göre kişinin rolü değişir.